lib/js/dom.nim

```diff
@@ -1460,8 +1460,8 @@ else:
   proc createTextNode*(d: Document, identifier: cstring): Node {.importcpp.}
   proc createComment*(d: Document, data: cstring): Node {.importcpp.}
 
-proc setTimeout*(action: proc(); ms: int): Timeout {.importc, nodecl.}
-proc clearTimeout*(t: Timeout) {.importc, nodecl.}
+proc setTimeout*(action: proc(); ms: int): TimeOut {.importc, nodecl.}
```

Comment:
That's not correct though..

Reply:
it's correct in the sense that it honors the style of the declaration:
```nim
TimeOut* {.importc.} = ref object of RootObj
```

but there are 2 conventions at play here: setTimeout and TimeOut, and since timeout is an actual word in the dictionary it should be Timeout, not TimeOut, so PR welcome to fix this via `s/TimeOut/Timeout/`

Reply:
> it's correct in the sense that it honors the style of the declaration:

Indeed - this PR was just to fix `styleCheck` complaints.

>  so PR welcome to fix this via s/TimeOut/Timeout/

The problem is that it's a breaking change to alter the style of exported symbols - it would break code that contains `TimeOut` and uses `--styleCheck:error`. The precedent set in https://github.com/nim-lang/Nim/pull/16176 is to keep bad style as-is.

Reply:
how about `-d:nimLegacyStyleTimeout` to resolve this issue, as done in previous instances:
```
changelog.md:52:23:  see #16034. Use `-d:nimLegacyReprWithNewline` for previous behavior.
changelog.md:67:50:  silently ignored errors.  See #16366.  Use `-d:nimLegacyEchoNoRaise` for previous behavior.
changelog.md:93:55:- Type mismatch errors now show more context, use `-d:nimLegacyTypeMismatch` for previous
compiler/types.nim:1494:24:    if conf.isDefined("nimLegacyTypeMismatch"):
lib/system/io.nim:786:35:  const echoDoRaise = not defined(nimLegacyEchoNoRaise) # see PR #16366
lib/system/repr.nim:328:18:    when defined(nimLegacyReprWithNewline): # see PR #16034
lib/system/reprjs.nim:240:16:  when defined(nimLegacyReprWithNewline): # see PR #16034
lib/system.nim:1338:16:    ## Use `-d:nimLegacyReprWithNewline` to revert to old behavior where newlines
tests/config.nims:16:19:switch("define", "nimLegacyTypeMismatch")
```

Reply:
> how about `-d:nimLegacyStyleTimeout`

I don't know.

But if we're considering adding another `-d`, we should consider letting that `-d` cover all the remaining problems that we can find, and then add `styleCheck` to CI so we don't introduce any new ones.

Reply:
ya, I've in meantime found a better idea, see https://github.com/nim-lang/Nim/pull/16177#issuecomment-749207787